mllib-dal/src/spark-3.1.1/main/scala/org/apache/spark/mllib/stat/Statistics.scala

```diff
@@ -0,0 +1,275 @@
+    * the chi-squared statistic is computed. All label and feature values must be categorical.
+    *
+    * @param data an `RDD[LabeledPoint]` containing the labeled dataset with categorical features.
+    *             Real-valued features will be treated as categorical for each distinct value.
+    * @return an array containing the ChiSquaredTestResult for every feature against the label.
+    *         The order of the elements in the returned array reflects the order of input features.
+    */
+  @Since("1.1.0")
+  def chiSqTest(data: RDD[LabeledPoint]): Array[ChiSqTestResult] = {
+    ChiSqTest.chiSquaredFeatures(data)
+  }
+
+  /**
+    * Java-friendly version of `chiSqTest()`
+    */
+  @Since("1.5.0")
+  def chiSqTest(data: JavaRDD[LabeledPoint]): Array[ChiSqTestResult] = chiSqTest(data.rdd)
+
+  /**
+    * Conduct the two-sided Kolmogorov-Smirnov (KS) test for data sampled from a
+    * continuous distribution. By comparing the largest difference between the empirical cumulative
+    * distribution of the sample data and the theoretical distribution we can provide a test for the
+    * the null hypothesis that the sample data comes from that theoretical distribution.
+    * For more information on KS Test:
+    * @see <a href="https://en.wikipedia.org/wiki/Kolmogorov%E2%80%93Smirnov_test">
+    * Kolmogorov-Smirnov test (Wikipedia)</a>
+    *
+    * @param data an `RDD[Double]` containing the sample of data to test
+    * @param cdf a `Double => Double` function to calculate the theoretical CDF at a given value
+    * @return [[org.apache.spark.mllib.stat.test.KolmogorovSmirnovTestResult]] object containing test
+    *        statistic, p-value, and null hypothesis.
+    */
+  @Since("1.5.0")
+  def kolmogorovSmirnovTest(data: RDD[Double], cdf: Double => Double)
+  : KolmogorovSmirnovTestResult = {
+    KolmogorovSmirnovTest.testOneSample(data, cdf)
+  }
+
+  /**
+    * Convenience function to conduct a one-sample, two-sided Kolmogorov-Smirnov test for probability
+    * distribution equality. Currently supports the normal distribution, taking as parameters
+    * the mean and standard deviation.
+    * (distName = "norm")
+    * @param data an `RDD[Double]` containing the sample of data to test
+    * @param distName a `String` name for a theoretical distribution
+    * @param params `Double*` specifying the parameters to be used for the theoretical distribution
+    * @return [[org.apache.spark.mllib.stat.test.KolmogorovSmirnovTestResult]] object containing test
+    *        statistic, p-value, and null hypothesis.
+    */
+  @Since("1.5.0")
+  @varargs
+  def kolmogorovSmirnovTest(data: RDD[Double], distName: String, params: Double*)
+  : KolmogorovSmirnovTestResult = {
+    KolmogorovSmirnovTest.testOneSample(data, distName, params: _*)
+  }
+
+  /**
+    * Java-friendly version of `kolmogorovSmirnovTest()`
+    */
+  @Since("1.5.0")
+  @varargs
+  def kolmogorovSmirnovTest(
```

Comment:
```suggestion
  def kolmogorovSmirnovTest(data: JavaDoubleRDD,
```
should format like this